lib/src/api/mod.rs

```diff
@@ -151,7 +151,11 @@ where
 			match conn.version().await {
 				Ok(version) => {
 					let server_build = &version.build;
-					if !req.matches(&version) {
+					if version.build.contains("dirty") {
```

Comment:
This feels more like bikeshedding but wouldn't `ends_with(".dirty")` be more "explicit" (for lack of a better word) here?

Reply:
It's a good point! I think we would prefer contains over ends_with, because there may be potentially unforeseen situations where something gets appended afterwards. But if a version contains "dirty" and it isn't actually dirty, then that would need to be addressed differently - perhaps then it could be changed to ends_with.

What do you think, does the above change your mind? Happy to change to ends_with, I think both approaches are valid. Contains will catch more scenarios, but having things appended is quite unlikely.

Reply:
Unless we change our `build.rs`, having something after `.dirty` is impossible https://github.com/surrealdb/surrealdb/blob/1e30eb4aa15b26a2ce5a36d539d42f85f6f9e753/build.rs#L33

Anyway, did you see https://github.com/surrealdb/surrealdb/pull/2259#issuecomment-1635773140 @phughk? After thinking about this a bit, I don't think we need to worry about this part of the codebase at all.